### ISSUE TITLE
Problems with Espeak if  trying to  run an old installer package

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3317]
**Reported by briang1 on 2013-07-01 12:03**
Installation of an older copy of the nvda installer over a current one fails with no speech if nvda is selected in the installed snap before loading the earlier installer.
In the case of  release versions, where errors are suppressed  it changes the synth to silence and gives no clues as to what is going on, but using a  last known bad snapshot you get the following log when you boot a working copy back in again.
nvda_snapshot_main-5967.exe
 gives the error, while...
nvda_snapshot_main-5973.exe does not and works correctly.

The error is 
INFO - synthDrivers.espeak.SynthDriver.**init** (12:24:03):
Using eSpeak version 1.46.02  06.Jan.12
WARNING - synthDriverHandler.SynthDriver.loadSettings (12:24:03):
No such voice: en
ERROR - synthDriverHandler.setSynth (12:24:03):
setSynth
Traceback (most recent call last):
  File "synthDriverHandler.pyc", line 80, in setSynth
  File "synthDriverHandler.pyc", line 453, in loadSettings
  File "synthDriverHandler.pyc", line 28, in changeVoice
  File "synthDrivers\espeak.pyc", line 153, in _set_voice
  File "synthDriverHandler.pyc", line 321, in _get_language
KeyError: u'en'
    DEBUG - speechDictHandler.SpeechDict.load (12:24:03):
Loading speech dictionary 'C:\Documents and Settings\brian\Application Data\nvda\speechDicts\silence.dic'...
DEBUG - speechDictHandler.SpeechDict.load (12:24:03):
file 'C:\Documents and Settings\brian\Application Data\nvda\speechDicts\silence.dic' not found.
INFO - synthDriverHandler.setSynth (12:24:03):
Loaded synthDriver silence

Now as this has obviously looked at the .ini file and found something it did not understand, maybe it just assumes its not available. Obviously, one cannot change something written back in time, but maybe  people could be warned somehow that its a bad idea to do this!

I also think that the save on exit should not be honoured in the installer itself, as this alters the .ini file and thus screws it up permanently to silence in this case on the last release version 2012.3.1. The snapshots are now fine in this regard. IE attempting to reload 2012.3.1, will end in silence  on the installed copy even if you back out of the installer or whatever. Best advice is to use a different synth before you attempt to load an old version, as this usually works.
However even if you do do this, afterwards any attempt to select 
espeak will fail of course. One would presumably need to completely uninstall the  later version before installing the older one.

I did this  test as a couple of people attempted this downdate,  and found it silent.

**Comment by bhavyashah** [comment]
CC @Qchristensen and a friendly ping to @jcsteh

**Comment by Qchristensen** [comment]
I'm not 100% I understand the original issue or whether it still applies.  I run old snapshots to test things all the time, with various synthesizers and haven't noticed a problem, (I don't generally run snapshots older than 2015 though).

In any case, as Jamie's comment noted, we can't go back and change old builds.  If anyone can identify if the issue still exists we could perhaps look at what we could fix going forward, otherwise I'd be inclined to close this.

**Comment by leonardder** [comment]
I"m closing this as per the suggestion in https://github.com/nvaccess/nvda/issues/3317#issuecomment-320195680
